### PR TITLE
Feature/add event public page api

### PR DESCRIPTION
**Pull request by dmitriy-uvin**
https://trello.com/c/dqnMrTI9/93-event-type-public-page-endpoints

Add Action, Request, Response for saving event.
Add method in controller for it.
Add migration to transform column type 'start_date' from string to timestamp.
Add EventRepository.
Add EventRequest.

### CODE REVIEW

File: api/app/Actions/Event/AddEventAction.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        $event->event_type_id = $request->getEventTypeId();
+        $event->invitee_name = $request->getInviteeName();
+        $event->invitee_email = $request->getInviteeEmail();
+        $event->start_date = $request->getStartDate();
+        $event->timezone = $request->getTimezone();
+
+        $this->eventRepository->save($event);
+
+        return new AddEventResponse();
```
Comment (aLLeXUs) [inline api/app/Actions/Event/AddEventAction.php:32]: The response is redundant. For the actions that do not return values, void type must be used.

File: api/app/Repositories/Event/EventRepository.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+namespace App\Repositories\Event;
+
+use App\Entity\Event;
+use App\Repositories\BaseRepository;
+
+final class EventRepository extends BaseRepository implements EventRepositoryInterface
+{
+    public function save(Event $event)
+    {
+        $event->save();
```
Comment (aLLeXUs) [inline api/app/Repositories/Event/EventRepository.php:14]: To be consistent with other repositories, the save/update method must return the saved entity.

File: api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('events', function (Blueprint $table) {
+            $table->dropColumn('start_date');
+        });
+        Schema::table('events', function (Blueprint $table) {
+            $table->timestamp('start_date');
```
Comment (aLLeXUs) [inline api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php:20]: You do not need to use a separate block for this.

File: api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
```
Comment (aLLeXUs) [inline api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php:30]: Reverse actions are missing.

File: api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php (hunk 3 above)
Comment (dmitriy-uvin) [reply api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php:20]: I think so, but i have an error with it.

Comment (aLLeXUs) [reply api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php:20]: Got it.

File: api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        Schema::table('events', function (Blueprint $table) {
+            $table->dropColumn('start_date');
```
Comment (aLLeXUs) [inline api/database/migrations/2020_08_10_194722_transform_start_date_to_timestamp.php:32]: Not all of the actions are reversing.

**Pull request merged**